src/engine/Service.cpp

```diff
@@ -142,19 +143,20 @@ ProtoResult Service::computeResultImpl([[maybe_unused]] bool requestLaziness) {
       serviceQuery, "application/sparql-query",
       "application/sparql-results+json");
 
-  std::basic_string<char, std::char_traits<char>,
-                    ad_utility::AllocatorWithLimit<char>>
-      jsonStr(_executionContext->getAllocator());
-  for (std::span<std::byte> bytes : response.body_) {
-    jsonStr.append(reinterpret_cast<const char*>(bytes.data()), bytes.size());
-    checkCancellation();
-  }
+  auto throwErrorWithContext = [&response, &serviceUrl](std::string_view sv) {
+    std::string ctx;
+    ctx.reserve(100);
+    for (const auto& s : std::move(response.body_)) {
```

Comment:
The `response` is a generator, which means in particular that `begin` or `for bla : generator` may only be called once, otherwise the whole engine might crash. 
We really must make sure, that this is the case whenever this function is called.
Maybe we find another way to make this more explicit.

Reply:
I see, you only use it very locally here, so this should work.